WeatherApp/WeatherApp/Services/WeatherService.swift

```diff
@@ -1,33 +1,73 @@
 import Foundation
 
 protocol WeatherServiceProtocol {
-    func fetchWeather(for cityName: String) async throws -> WeatherModel
+
+    func fetchWeather(for cityName: String, completion: @escaping (Result<CurrentWeatherResponse, ClientError>) -> Void)
+
 }
 
 class WeatherService: WeatherServiceProtocol {
 
+    private let client: BaseApiClientProtocol
+    private let endpointFactory: WeatherEndpointFactory
+
     private let baseURL = "https://api.openweathermap.org/data/2.5/weather"
-    private var apiKey: String {
-            guard let apiKey = Bundle.main.infoDictionary?["API_KEY"] as? String else {
-                return "API Key not found"
-            }
-            return apiKey
+
+    init(client: BaseApiClientProtocol) {
+        self.client = client
+
+        guard
+            let apiKey = InfoConstants.openWeatherMapApiKey
+        else {
+            fatalError("API Key not found")
+        }
```

Comment:
could be in the same line if it fits
`       guard let apiKey = InfoConstants.openWeatherMapApiKey else { fatalError("API Key not found") }
`       
       else:
       
```swift
       guard let apiKey = InfoConstants.openWeatherMapApiKey else { 
           fatalError("API Key not found")
        }
```